internal/renderers/skia/lib.rs

```diff
@@ -83,6 +86,99 @@ fn create_default_surface(
     }
 }
 
+#[derive(Default)]
+struct PartialRenderingState {
+    partial_cache: RefCell<i_slint_core::item_rendering::PartialRenderingCache>,
+    /// This is the area which we are going to redraw in the next frame, no matter if the items are dirty or not
+    force_dirty: RefCell<DirtyRegion>,
+    repaint_buffer_type: Cell<RepaintBufferType>,
+    /// This is the area which was dirty on the previous frame.
+    /// Only used if repaint_buffer_type == RepaintBufferType::SwappedBuffers
+    prev_frame_dirty: Cell<DirtyRegion>,
+    /// Force a redraw in the next frame, no matter what's dirty. Use only as a last resort.
+    force_screen_refresh: Cell<bool>,
+}
+
+impl PartialRenderingState {
+    fn set_repaint_buffer_type(&self, repaint_buffer_type: RepaintBufferType) {
+        if self.repaint_buffer_type.replace(repaint_buffer_type) != repaint_buffer_type {
+            self.partial_cache.borrow_mut().clear();
+        }
+    }
+
+    fn create_partial_renderer<'a, T: ItemRenderer>(
+        &'a self,
+        renderer: T,
+    ) -> i_slint_core::item_rendering::PartialRenderer<'a, T> {
+        i_slint_core::item_rendering::PartialRenderer::new(
+            &self.partial_cache,
+            self.force_dirty.take(),
+            renderer,
+        )
+    }
+
+    fn apply_dirty_region<T: ItemRenderer>(
```

Comment:
I guess this was the "code duplication" you were talking about :-)

Reply:
Yeahh